moralization/analyse.py

```diff
@@ -71,14 +73,55 @@ def sort_spans(cas: object, ts: object) -> defaultdict:
     return span_dict
 
 
-def get_sentences(cas: object, ts: object) -> list:
+def add_true_sentences(cas: object, ts: object) -> object:
+    nlp = spacy.load("de_core_news_sm")
+    text = cas.sofa_string
+    sentences = [sentence for sentence in nlp(text).sents]
+    # create new typesystem:
+    try:
+        True_sentence_type = ts.create_type(name="moralization.TrueSentence")
+    except ValueError:
+        True_sentence_type = ts.get_type("moralization.TrueSentence")
+
+    for sentence in sentences:
+        if (sentence.end - sentence.start) > 3:
+            # spacy returns the token ids, not the character ids.
+            # this is in missmatch with cassis.
+            # to convert from token to character I search through the doc with a regex to find the correct ids.
+            # the replace is to avoid regex errors in regards to brackets
```

Comment:
I'm confused by this - does spacy not allow you to see the location of a token, e.g. `token.idx`?

Reply:
My mistake, i first only found the `token.start`and `token.end` variables.
but spacy also has `token.start_char`which does exactly what I needed.